test/extensions/stats_sinks/statsd/config_test.cc

```diff
@@ -45,6 +45,101 @@ TEST(StatsConfigTest, ValidTcpStatsd) {
   EXPECT_NE(dynamic_cast<Common::Statsd::TcpStatsdSink*>(sink.get()), nullptr);
 }
 
+TEST(StatsConfigTest, UdpSinkDefaultPrefix) {
+  const std::string name = StatsSinkNames::get().STATSD;
+  auto defaultPrefix = Common::Statsd::getDefaultPrefix();
+
+  envoy::config::metrics::v2::StatsdSink sink_config;
+  envoy::api::v2::core::Address& address = *sink_config.mutable_address();
+  envoy::api::v2::core::SocketAddress& socket_address = *address.mutable_socket_address();
+  socket_address.set_protocol(envoy::api::v2::core::SocketAddress::UDP);
+  socket_address.set_address("127.0.0.1");
+  socket_address.set_port_value(8125);
+  EXPECT_EQ(sink_config.prefix(), "");
+
+  Server::Configuration::StatsSinkFactory* factory =
+      Registry::FactoryRegistry<Server::Configuration::StatsSinkFactory>::getFactory(name);
+  ASSERT_NE(factory, nullptr);
+  ProtobufTypes::MessagePtr message = factory->createEmptyConfigProto();
+  MessageUtil::jsonConvert(sink_config, *message);
+
+  NiceMock<Server::MockInstance> server;
+  Stats::SinkPtr sink = factory->createStatsSink(*message, server);
+  EXPECT_NE(sink, nullptr);
+  EXPECT_NE(dynamic_cast<Common::Statsd::UdpStatsdSink*>(sink.get()), nullptr);
```

Comment:
Two comments:
1.  Same as above, I think this should be `ASSERT_NE` since you'll get a segfault below if it's `nullptr`.
2.  nit: this is a little hard to read - can we just dynamic cast once, save to a local variable, and then use the resulting pointer to do both checks?

(Same comments for other test cases)

Reply:
Fixed in all four tests.